siliconcompiler/tools/surelog/surelog.py

```diff
@@ -101,18 +101,44 @@ def runtime_options(chip):
 
     cmdlist = []
 
-    # source files
-    for value in chip.find_files('option', 'ydir'):
+    # Deduplicated source files
+    # Library directories.
+    ydir_files = chip.find_files('option', 'ydir')
+    if ydir_files != list(set(ydir_files)):
```

Comment:
I think `set()` isn't guaranteed to preserve order, so I'm not sure if this always works. How about something like 

```python
if len(ydir_files) != len(set(ydir_files)):
  ... 
```

Reply:
Good point, thanks.